public/app/plugins/datasource/loki/configuration/DerivedField.test.tsx

```diff
@@ -1,68 +1,93 @@
+  beforeEach(() => {
+    setDataSourceSrv({
+      get: jest.fn(),
+      reload: jest.fn(),
+      getInstanceSettings: jest.fn(),
+      getList: mockList.mockImplementation(() => [
+        {
+          id: 1,
+          uid: 'metrics',
+          name: 'metrics_ds',
+          meta: {
+            tracing: false,
+            info: {
+              logos: {
+                small: '',
+              },
+            },
+          } as DataSourcePluginMeta,
+        } as DataSourceInstanceSettings,
+        {
+          id: 2,
+          uid: 'tracing',
+          name: 'tracing_ds',
+          meta: {
+            tracing: true,
+            info: {
+              logos: {
+                small: '',
+              },
+            },
+          } as DataSourcePluginMeta,
+        } as DataSourceInstanceSettings,
+      ]),
+    });
+  });
+
+  it('shows internal link if uid is set', async () => {
     const value = {
       matcherRegex: '',
       name: '',
       datasourceUid: 'test',
     };
-    const wrapper = shallow(<DerivedField value={value} onChange={() => {}} onDelete={() => {}} suggestions={[]} />);
+    // Render and wait for the Name field to be visible
+    // using findBy to wait for asynchronous operations to complete
+    render(<DerivedField value={value} onChange={() => {}} onDelete={() => {}} suggestions={[]} />);
+    expect(await screen.findByText('Name')).toBeInTheDocument();
 
-    expect(wrapper.find(DataSourcePicker).length).toBe(1);
+    expect(screen.getByLabelText(selectors.components.DataSourcePicker.inputV2)).toBeInTheDocument();
   });
 
-  it('shows url link if uid is not set', () => {
+  it('shows url link if uid is not set', async () => {
     const value = {
       matcherRegex: '',
       name: '',
       url: 'test',
     };
-    const wrapper = shallow(<DerivedField value={value} onChange={() => {}} onDelete={() => {}} suggestions={[]} />);
-    expect(wrapper.find(DataSourcePicker).length).toBe(0);
+    // Render and wait for the Name field to be visible
+    // using findBy to wait for asynchronous operations to complete
+    render(<DerivedField value={value} onChange={() => {}} onDelete={() => {}} suggestions={[]} />);
+    expect(await screen.findByText('Name')).toBeInTheDocument();
+
+    expect(screen.queryByLabelText(selectors.components.DataSourcePicker.inputV2)).not.toBeInTheDocument();
   });
 
-  it('shows only tracing datasources for internal link', () => {
+  it('shows only tracing datasources for internal link', async () => {
     const value = {
       matcherRegex: '',
       name: '',
       datasourceUid: 'test',
     };
-    const wrapper = shallow(<DerivedField value={value} onChange={() => {}} onDelete={() => {}} suggestions={[]} />);
-    expect(wrapper.find(DataSourcePicker).props().tracing).toEqual(true);
+    // Render and wait for the Name field to be visible
+    // using findBy to wait for asynchronous operations to complete
```

Comment:
Not really sure if repeating the message 3 times is desirable, although I understand the intention. If I had to make a suggestion, I would either not use it at all or maybe just write it once? Nickpick anyway.

Reply:
yeah that's fair, i went with being overly verbose in case tests get moved around etc 🤷